File: src/main/java/coza/trojanc/stonewriter/context/impl/DefaultContextMap.java
package coza.trojanc.stonewriter.context.impl;

import coza.trojanc.stonewriter.context.ContextMap;

import java.util.HashMap;
import java.util.Map;


/**
 * The type Default context map.
 */
public class DefaultContextMap implements ContextMap {

	private final Map<String, String> resolvedVariables;

	/**
	 * Instantiates a new Default context map.
	 */
	public DefaultContextMap(){
		this(0);
	}

	/**
	 * Instantiates a new Default context map.
	 *
	 * @param initialSize the initial size
	 */
	public DefaultContextMap(int initialSize){
		resolvedVariables = new HashMap<>(initialSize);
	}

	@Override
	public void add(String name, String value){
		resolvedVariables.put(name, value);
	}

	@Override
	public String get(String name){
		return resolvedVariables.get(name);
	}

	@Override
	public boolean has(String key) {
		return resolvedVariables.containsKey(key);
	}
}

File: receipt-shared/src/main/kotlin/coza/trojanc/receipt/context/impl/DefaultContextMap.kt
package coza.trojanc.receipt.context.impl

import coza.trojanc.receipt.context.ContextMap

/**
 * Default implementation of a [ContextMap]
 * @author Charl Thiem
 */
class DefaultContextMap : ContextMap {

    /**
     * Instantiates a new Default context map.
     */
    constructor(){
        resolvedVariables = hashMapOf();
    }

    private val resolvedVariables: MutableMap<String, String>

    override fun add(key: String, value: String) {
        resolvedVariables.put(key, value)
    }

    override fun get(key: String): String? {
        return resolvedVariables.get(key)
    }

    override fun has(key: String): Boolean {
        return resolvedVariables.containsKey(key)
    }
}
